packages/harmony/src/components/button/FollowButton/FollowButton.tsx

```diff
@@ -55,66 +60,81 @@ export const FollowButton = (props: FollowButtonProps) => {
     setIsHovering(false)
   }, [setIsHovering])
 
-  const handleChange = useCallback(() => {
+  const handleValueChange = useCallback(() => {
     if (value) {
       onUnfollow?.()
     } else {
       onFollow?.()
     }
     setValueState(!value)
-  }, [value, setValueState, onFollow, onUnfollow])
+  }, [value, setValueState, onUnfollow, onFollow])
+
+  const handleClick = () => {
+    handleValueChange()
+  }
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    onChange?.(e)
+    handleValueChange()
+  }
 
+  const checkedValue = value || checked
   let Icon: IconComponent | null = IconUserFollow
   let text = messages.follow
-  if (value && !isHovering) {
+  if (checkedValue && !isHovering) {
     Icon = IconUserFollowing
     text = messages.following
-  } else if (value && isHovering) {
+  } else if (checkedValue && isHovering) {
     Icon = IconUserUnfollow
     text = messages.unfollow
   }
 
   const { color, cornerRadius } = useTheme()
   const textColor =
-    value || isHovering ? color.static.white : color.primary.primary
+    checkedValue || isHovering ? color.static.white : color.primary.primary
   const css: CSSObject = {
+    minWidth: size === 'small' ? 128 : 152,
     userSelect: 'none',
     border: `1px solid ${color.primary.primary}`,
     borderRadius: variant === 'pill' ? cornerRadius['2xl'] : cornerRadius.s,
     background:
-      value || isHovering ? color.primary.primary : color.static.white,
+      checkedValue || isHovering ? color.primary.primary : color.static.white,
     ':active': {
       background: color.primary.p500,
       border: `1px solid ${color.primary.p500}`
     }
   }
 
   return (
-    <>
-      <label onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
-        <InputRoot type='checkbox' onChange={handleChange} {...inputProps} />
-        <Flex
-          w={size === 'small' ? 128 : 152}
-          h={size === 'small' ? 28 : 32}
-          direction='row'
-          alignItems='center'
-          justifyContent='center'
-          gap='xs'
-          pv='s'
-          css={css}
+    <label onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
```

Comment:
so we shouldnt have label for case when its a button. either make that a div, or follow what we did with selectable-pill where the outer container becomes button